### PR TITLE
Fix user quota should be calculated based on group

### PR DESCRIPTION
**Pull request by jackklpan**
Signed-off-by: Jack Pan <jackpan@infuseai.io>

** PR checklist **

- [*] Ensure you have added or ran the appropriate tests for your PR.
- [*] DCO signed

**What type of PR is this?**
Fix bug.

**What this PR does / why we need it**:
The user quota is calculated based on a group.

**Which issue(s) this PR fixes**:
Fix user quota calculation bug.

**Special notes for your reviewer**:
No

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: controllers/phjob_scheduler.go [hunk 1]
```diff
@@ -107,12 +108,19 @@ func (r *PHJobScheduler) getInstanceTypeInfo(instanceTypeId string) (*graphql.Dt
 	return InstanceTypeCache.Get(cacheKey).Value().(*graphql.DtoInstanceType), nil
 }
 
-func (r *PHJobScheduler) getCurrentUsage(namespace string, aggregation_key string, aggregation_value string) (*ResourceQuota, error) {
+func (r *PHJobScheduler) getCurrentUsage(namespace string, aggregation_keys []string, aggregation_values []string) (*ResourceQuota, error) {
+	if len(aggregation_keys) != len(aggregation_values) {
+		return NewResourceQuota(), errors.New("length of key and value must be the same")
```
Comment (qrtt1) [inline controllers/phjob_scheduler.go]: should it be ?

```go
return nil, errors.New("length of key and value must be the same")
```

File: controllers/phjob_scheduler.go [hunk 2]
```diff
@@ -291,7 +299,7 @@ func (r *PHJobScheduler) scheduleByStrictOrder(phJobsRef *[]*primehubv1alpha1.Ph
 
 		_, foundUser := usersRemainingQuota[phJob.Spec.UserName]
 		if !foundUser {
-			userRemainingQuota, err := r.getUserRemainingQuota(phJob)
+			userRemainingQuota, err := r.getUserRemainingQuotaInGroup(phJob)
 			if err != nil {
 				return err
```
Comment (qrtt1) [inline controllers/phjob_scheduler.go]: should it be ?

```go
return nil, err 
```

File: controllers/phjob_scheduler.go (hunk 1 above)
Comment (jackklpan) [reply controllers/phjob_scheduler.go]: Thx. Fixed!

File: controllers/phjob_scheduler.go (hunk 2 above)
Comment (jackklpan) [reply controllers/phjob_scheduler.go]: Here we just return err based on scheduleByStrictOrder function.

File: controllers/phjob_scheduler.go [hunk 3]
```diff
@@ -157,12 +165,12 @@ func (r *PHJobScheduler) getGroupRemainingQuota(phJob *primehubv1alpha1.PhJob) (
...
 }
 
-func (r *PHJobScheduler) getUserRemainingQuota(phJob *primehubv1alpha1.PhJob) (*ResourceQuota, error) {
+func (r *PHJobScheduler) getUserRemainingQuotaInGroup(phJob *primehubv1alpha1.PhJob) (*ResourceQuota, error) {
 	groupInfo, err := r.getGroupInfo(phJob.Spec.GroupId)
 	if err != nil {
 		return nil, err
 	}
-	userUsage, err := r.getCurrentUsage(phJob.Namespace, user_aggregation_key, phJob.Spec.UserName)
+	userUsage, err := r.getCurrentUsage(phJob.Namespace, []string{group_aggregation_key, user_aggregation_key}, []string{phJob.Spec.GroupName, phJob.Spec.UserName})
```
Comment (qrtt1) [inline controllers/phjob_scheduler.go]: group_aggregation_key and user_aggregation_key should be CamelCase

Comment (jackklpan) [reply controllers/phjob_scheduler.go]: Fixed. Thanks!

**Pull request merged**